### PR TITLE
FEM-2502 AndroidX migration + ExoPlayer upgrade

### PR DESCRIPTION
**Pull request by GouravSna**
- ExoPlayer upgraded to 2.10.1
- Decoder reusability on change media (setForegroundMode())
- Support library migrated to AndroidX.

### CODE REVIEW

File: playkit/build.gradle [hunk 1]
```diff
@@ -35,9 +35,9 @@ tasks.withType(Javadoc) {
 
 dependencies {
 
-    def exoPlayerVersion = '2.9.6'
+    def exoPlayerVersion = '2.10.1'
 
-    api 'com.google.code.gson:gson:2.8.2'
+    api 'com.google.code.gson:gson:2.8.5'
```
Comment (noamtamim) [inline playkit/build.gradle]: Keep at 2.8.2

File: playkit/build.gradle [hunk 2]
```diff
@@ -46,13 +46,14 @@ dependencies {
     api "com.google.android.exoplayer:exoplayer-ui:$exoPlayerVersion"
     api "com.google.android.exoplayer:extension-okhttp:$exoPlayerVersion"
 
+    implementation 'androidx.annotation:annotation:1.0.2'
+
     // Ok is (optionally) used by ExoPlayer now
-    api("com.squareup.okhttp3:okhttp:3.12.1")
+    api('com.squareup.okhttp3:okhttp:3.14.2')
```
Comment (noamtamim) [inline playkit/build.gradle]: Keep as 3.12.1

File: playkit/src/main/java/com/kaltura/playkit/player/PKExternalSubtitle.java [hunk 3]
```diff
@@ -14,6 +15,11 @@
     private String id;
     private String mimeType;
     private int selectionFlags = Consts.TRACK_UNSELECTED_FLAG;
+    /**
+     * Indicates the track contains subtitles. This flag may be set on video tracks to indicate the
+     * presence of burned in subtitles.
+     */
+    private int roleFlag = C.ROLE_FLAG_SUBTITLE;
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/player/PKExternalSubtitle.java:22]: Where is this field used?

File: playkit/src/main/java/com/kaltura/playkit/player/PKExternalSubtitle.java [hunk 4]
```diff
@@ -64,6 +70,10 @@ public boolean isDefault() {
         return isDefault;
     }
 
+    public int getRoleFlag() {
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/player/PKExternalSubtitle.java:73]: Where is this getter used?

**Review by noamtamim — CHANGES_REQUESTED** [review]

File: playkit/src/main/java/com/kaltura/playkit/player/PKExternalSubtitle.java (hunk 3 above)
Comment (GouravSna) [reply playkit/src/main/java/com/kaltura/playkit/player/PKExternalSubtitle.java:22]: Fixed. It is being used in `ExoPlayerWrapper` in `buildExternalSubtitleSource`

File: playkit/build.gradle (hunk 2 above)
Comment (GouravSna) [reply playkit/build.gradle]: Updated.

File: playkit/build.gradle (hunk 1 above)
Comment (GouravSna) [reply playkit/build.gradle]: Updated

**Review by noamtamim — APPROVED** [review]

**Pull request merged**